.dev/compare_branches.R

```diff
@@ -207,164 +223,127 @@ get_deps <- function(pkg) {
-  pkg_is_dir <- file.info(pkgs)$isdir
-  pkg_names <- dplyr::if_else(
-    pkg_is_dir,
-    basename(pkgs),
-    gsub("_.*", "", basename(pkgs))
-  )
-
-  # given how common it is to skip packages (e.g. due to uninstalled
-  #   dependencies), use a while loop to try and reach n_packages instead
-  #   of just iterating over n_packages (which may in actuality lint
-  #   far fewer than that number)
-  lints <- vector("list", n_packages)
-  lint_names <- character(n_packages)
-  ii <- 1L
-  jj <- 0L
-  while (ii <= length(pkgs) && jj <= n_packages) {
-    if (pkg_is_dir[ii]) {
-      pkg <- pkgs[ii]
-    } else {
-      tmp <- file.path(tempdir(), pkg_names[ii])
-      on.exit(unlink(tmp, recursive = TRUE))
-      # --strip-components makes sure the output structure is
-      # /path/to/tmp/pkg/ instead of /path/to/tmp/pkg/pkg
-      utils::untar(pkgs[ii], exdir = tmp, extras = "--strip-components=1")
-      pkg <- tmp
-    }
-    if (test_encoding(pkg)) {
-      if (warn) {
-        warning(sprintf(
-          "Package %s has some files with unknown encoding; skipping",
-          pkg_names[ii]
-        ))
-      }
-      ii <- ii + 1L
-      next
-    }
-    # object_usage_linter requires running package code, which may
-    #   not work if the package has unavailable Depends;
-    # object_name_linter also tries to run loadNamespace on Imports
-    #   found in the target package's NAMESPACE file
-    if (check_depends) {
-      pkg_deps <- get_deps(pkg)
-      if ("tcltk" %in% pkg_deps && !capabilities("tcltk")) {
-        if (warn) {
-          warning(sprintf(
-            "Package %s depends on tcltk, which is not available (via capabilities()); skipping",
-            pkg_names[ii]
-          ))
-        }
-        ii <- ii + 1L
-        next
-      }
-      try_deps <- tryCatch(
-        find.package(pkg_deps),
-        error = identity,
-        warning = identity
-      )
-      if (inherits(try_deps, c("warning", "error"))) {
-        if (warn) {
-          warning(sprintf(
-            "Some package Dependencies for %s were unavailable: %s; skipping",
-            pkg_names[ii],
-            gsub("there (?:are no packages|is no package) called ", "", try_deps$message)
-          ))
-        }
-        ii <- ii + 1L
-        next
-      }
-    }
-    jj <- jj + 1L
-    lints[[jj]] <- lint_dir(pkg, linters = linter, parse_settings = FALSE)
-    lint_names[jj] <- pkg_names[ii]
-    ii <- ii + 1L
+lint_one_package <- function(package, linters, out_dir, check_deps) {
+  package_is_dir <- file.info(package)$isdir
+  package_name <- gsub("_.*", "", basename(package))
+
+  if (!package_is_dir) {
+    tmp <- file.path(tempdir(), package_name)
+    # only extract files that lintr::lint_package() cares about
+    package_files <- utils::untar(package, list = TRUE)
+    lint_files <- grep(file.path(package_name, "(R|tests|inst|vignettes|data-raw|demo)"), package_files, value = TRUE)
```

Comment:
Shouldn't this regex be anchored just in case?

Reply:
Hmm I actually don't think this is used. I guess I never solved how to extract only certain files here. Switched to a TODO instead.